src/collection.hpp

```diff
@@ -0,0 +1,81 @@
+// Unless explicitly stated otherwise all files in this repository are
+// dual-licensed under the Apache-2.0 License or BSD-3-Clause License.
+//
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2021 Datadog, Inc.
+
+#pragma once
+
+#include <event.hpp>
+#include <rule.hpp>
+
+#include <unordered_map>
+#include <unordered_set>
+#include <vector>
+
+namespace ddwaf {
+
+// The collection cache is shared by both priority and regular collections,
+// this ensures that regular collections aren't processed when the respective
+// priority collection has already had a match.
```

Comment:
I'm not sure I understand this comment (or the related one when setting `cache.result = true`) in the implementation file. What is the "respective" (the language here) or the "equivalent" (in the .cpp file) regular collection of a priority collection? As far as I can see, they are not created in pairs. Are you just talking about when priority collections act as regular collections (when all the actions were already seen)?

Reply:
A priority collection is created for rules with `on_match`, while a regular (i.e. non-priority) collection is created for rules without `on_match`, if there are rules with the same `type` field with and without `on_match` there will be one priority collection for `type` and one regular collection for `type`. However, if there is a match in the priority collection for `type` we should skip corresponding regular collection. The existence of a pair is not guaranteed, as it depends on the ruleset.

I can clarify this comment.